docs/csharp/language-reference/statements/declarations.md

```diff
@@ -0,0 +1,132 @@
+
+```csharp
+var xs = new List<int>();
+```
+
+Beginning with C# 9.0, you can use a target-typed [`new` expression](../operators/new-operator.md) as an alternative:
+
+```csharp
+List<int> xs = new();
+List<int>? ys = new();
+```
+
+In pattern matching, the `var` keyword is used in a [`var` pattern](../operators/patterns.md#var-pattern).
+
+The following example shows two query expressions. In the first expression, the use of `var` is permitted but isn't required, because the type of the query result can be stated explicitly as an `IEnumerable<string>`. However, in the second expression, `var` allows the result to be a collection of anonymous types, and the name of that type isn't accessible except to the compiler itself. Use of `var` eliminates the requirement to create a new class for the result. In Example #2, the `foreach` iteration variable `item` must also be implicitly typed.
+
+:::code language="csharp" source="./snippets/declarations/ImplicitlyTyped.cs" id="VarExample":::
+
+## Ref locals
+
+You add the `ref` keyword before the type of a variable to declare a `ref` local. Assume the `GetContactInformation` method is declared as a [ref return](jump-statements.md#ref-returns):
+
+```csharp
+public ref Person GetContactInformation(string fname, string lname)
+```
+
+A by-value assignment reads the value of a variable and assigns it to a new variable:
+
+```csharp
+Person p = contacts.GetContactInformation("Brandie", "Best");
+```
+
+The preceding assignment declares `p` as a local variable. Its initial value is copied from reading the value returned by `GetContactInformation`. Any future assignments to `p` won't change the value of the variable returned by `GetContactInformation`. The variable `p` is no longer an alias to the variable returned.
+
+You declare a *ref* variable to copy the alias to the original value. In the following assignment, `p` is an alias to the variable returned from `GetContactInformation`.
+
+```csharp
+ref Person p = ref contacts.GetContactInformation("Brandie", "Best");
+```
+
+Subsequent usage of `p` is the same as using the variable returned by `GetContactInformation` because `p` is an alias for that variable. Changes to `p` also change the variable returned from `GetContactInformation`.
+
+You can access a value by reference in the same way. In some cases, accessing a value by reference increases performance by avoiding a potentially expensive copy operation. For example, the following statement shows how one can define a ref local value that is used to reference a value.
+
+```csharp
+ref VeryLargeStruct reflocal = ref veryLargeStruct;
+```
+
+The `ref` keyword is used both before the local variable declaration *and* before the value in the second example. Failure to include both `ref` keywords in the variable declaration and assignment in both examples results in compiler error CS8172, "Can't initialize a by-reference variable with a value."
```

Comment:
Could we link to compiler error CS8172 here?

Reply:
not yet. We don't have help on that one yet.